apps/dashboard/src/components/workflow-editor/channel-preferences-form.tsx

```diff
@@ -0,0 +1,305 @@
+import { useMemo } from 'react';
+import { Link } from 'react-router-dom';
+import { motion } from 'motion/react';
+import { useForm, useWatch } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { RiArrowLeftSLine, RiCloseFill, RiInformationFill } from 'react-icons/ri';
+import { ChannelTypeEnum, UpdateWorkflowDto, WorkflowPreferences, WorkflowResponseDto } from '@novu/shared';
+
+import { SidebarContent, SidebarHeader } from '@/components/side-navigation/sidebar';
+import { UserPreferencesFormSchema } from '@/components/workflow-editor/schema';
+import { STEP_TYPE_TO_COLOR } from '@/utils/color';
+import { StepTypeEnum, WorkflowOriginEnum } from '@/utils/enums';
+import { capitalize } from '@/utils/string';
+import { cn } from '@/utils/ui';
+import { STEP_TYPE_TO_ICON } from '../icons/utils';
+import { PageMeta } from '../page-meta';
+import { Button } from '../primitives/button';
+import { Card, CardContent } from '../primitives/card';
+import { Checkbox } from '../primitives/checkbox';
+import { Form, FormControl, FormField, FormItem, FormLabel } from '../primitives/form/form';
+import { Separator } from '../primitives/separator';
+import { Step } from '../primitives/step';
+import { Switch } from '../primitives/switch';
+import { useTelemetry } from '@/hooks';
+import { TelemetryEvent } from '@/utils/telemetry';
+
+type ConfigureWorkflowFormProps = {
+  workflow: WorkflowResponseDto;
+  update: (data: UpdateWorkflowDto) => void;
+};
+
+const CHANNEL_TYPE_TO_STRING: Record<ChannelTypeEnum, string> = {
+  [ChannelTypeEnum.IN_APP]: 'In-App',
+  [ChannelTypeEnum.EMAIL]: 'Email',
+  [ChannelTypeEnum.SMS]: 'SMS',
+  [ChannelTypeEnum.CHAT]: 'Chat',
+  [ChannelTypeEnum.PUSH]: 'Push',
+};
+
+const CHANNEL_LABELS_LOOKUP: Record<`${ChannelTypeEnum}` | 'all', string> = {
+  ...CHANNEL_TYPE_TO_STRING,
+  all: 'All',
+};
+
+const checkIsEveryChannelSameValue = (
+  channels: Record<ChannelTypeEnum, { enabled: boolean }>,
+  checkForEnabled: boolean
+) => {
+  return Object.values(channels).every((channel) => channel.enabled === checkForEnabled);
+};
+
+export const ChannelPreferencesForm = (props: ConfigureWorkflowFormProps) => {
+  const { workflow, update } = props;
+  const track = useTelemetry();
+
+  const { formDataToRender, isDefaultPreferences, isDashboardWorkflow } = useMemo(() => {
```

Comment:
This memo is a bit overloaded. Consider splitting to three separate memos with simpler dependencies.